### PR TITLE
[WASI-NN] piper: fix piper dependencies

### PR DESCRIPTION
**Pull request by PeterD1524**
Hi, I am working on [LFX Workspace: Support piper as a new backend of the WASI-NN WasmEdge plugin #3483](https://www.github.com/WasmEdge/WasmEdge/issues/3483)

The dependencies of WASI-NN Piper plugin:

Currently the dependency piper unconditionally creates (or downloads) shared libraries of its dependencies:

libwasmedgePluginWasiNN.so -> libpiper.a -> libpiper_phonemize.so -> (libespeak-ng.so, libonnxruntime.so)

As a result, libwasmedgePluginWasiNN.so cannot work on its own.
This PR attempts to solve these problems.

The final result will be:

libwasmedgePluginWasiNN.so -> libpiper.a -> libpiper_phonemize.a -> (libespeak-ng.a, system libonnxruntime.so)

The plugin will static link to most of its dependencies and it will dynamic link to a system provided libonnxruntime.so.
Users will have to install onnxruntime themselves.

### CODE REVIEW

**Comment by juntao** [comment]
Hello, I am a [code review agent](https://github.com/flows-network/github-pr-review/) on [flows.network](https://flows.network/).

It could take a few minutes for me to analyze this PR. Relax, grab a cup of coffee and check back later. Thanks!

File: .github/workflows/build-extensions.yml [hunk 1]
```diff
@@ -116,6 +116,7 @@ jobs:
           bash utils/wasi-nn/install-openvino.sh
           bash utils/wasi-nn/install-pytorch.sh
           bash utils/wasi-nn/install-neuralspeed.sh
+          bash utils/wasi-nn/install-onnxruntime-ubuntu.sh
```
Comment (0yi0) [inline .github/workflows/build-extensions.yml:119]: I think we don't need `-ubuntu` suffix here? @hydai how do you think?

Comment (hydai) [reply .github/workflows/build-extensions.yml:119]: We should remove the suffix because this should also be used in the manylinux(CentOS) distributions.

**Review by hydai — APPROVED** [review]
LGTM

**Pull request merged**